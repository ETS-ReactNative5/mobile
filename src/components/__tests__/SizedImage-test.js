import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import SizedImage from '../SizedImage'

jest.useFakeTimers();

jest.mock('Image', () => {
  const React = require('react');
  class MockImage extends React.Component {

    static getSize() {
      return 40;
    }
  
    render() {
      return null;
    }
  }
  return MockImage
});

const source = {
  uri: 'https://placekitten.com/200/300'
}

it('renders', () => {
  const tree = renderer.create(
    <SizedImage
      source={source}
      additionalStyles={[]}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
