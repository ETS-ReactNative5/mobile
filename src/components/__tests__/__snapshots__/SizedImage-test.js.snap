exports[`test renders 1`] = `
<Image
  defaultSource={1}
  onLoadEnd={[Function]}
  source={
    Object {
      "uri": "https://placekitten.com/200/300",
    }
  }
  style={
    Object {
      "height": 0,
      "opacity": 0,
      "width": 0,
    }
  } />
`;
